packages/signals/signals-integration-tests/src/tests/signals-vanilla/signals-ingestion.test.ts

```diff
@@ -27,3 +27,50 @@ test('ingestion enabled -> will send the signal', async ({ page }) => {
 
   expect(true).toBe(true)
 })
+
+test('never sends signal when sample rate is 0', async ({ page }) => {
+  await indexPage.loadAndWait(
+    page,
+    basicEdgeFn,
+    {
+      flushAt: 1,
+      enableSignalsIngestion: false,
+    },
+    undefined,
+    {
+      edgeFunction: {
+        downloadURL: 'https://cdn.edgefn.segment.com/MY-WRITEKEY/foo.js',
+        version: 1,
+      },
+      autoInstrumentationSettings: {
+        sampleRate: 0,
+      },
+    }
+  )
+  await indexPage.fillNameInput('John Doe')
+  await page.waitForTimeout(100)
+  expect(indexPage.signalsAPI.getEvents('interaction')).toHaveLength(0)
+})
+
+test('always sends signal when sample rate is 1', async ({ page }) => {
+  await indexPage.loadAndWait(
+    page,
+    basicEdgeFn,
+    {
+      flushAt: 1,
+      enableSignalsIngestion: false,
+    },
+    undefined,
+    {
+      edgeFunction: {
+        downloadURL: 'https://cdn.edgefn.segment.com/MY-WRITEKEY/foo.js',
```

Comment:
See comment about not needing to pass the entire cdnSettings -- just overriding the bits we care about